### PR TITLE
 mainnet tokens autopopulate add token form fields when on custom networks #12087

### LINKED ISSUE #12087: mainnet tokens autopopulate add token form fields when on custom networks
1. Be on a custom network
2. Copy the mainnet address of a popular erc20 token contract
3. Go to the add token flow, paste the erc20 contract address
4. The decimal and symbol fields will populate with mainnet data and the user will be able to add the token

Instead, we should show the user the same warnings we show when they paste a non-contract address.

This might be resolved with our upcoming token detection improvements.

### PR DESCRIPTION
**Pull request by dragana8**
Fixes: https://github.com/MetaMask/metamask-extension/issues/12087

Explanation:  
When trying to import mainnet token user gets error message 'This is a mainnet token' and symbol and decimals fields are not autopopulated but user is able to submit.
Screenshot from 2021-11-23 11-57-18

### CODE REVIEW

**Comment by mhasanagic** [comment]
Verified by QA

File: ui/pages/import-token/import-token.component.js [hunk 1]
```diff
@@ -191,13 +199,24 @@ class ImportToken extends Component {
...
     });
     const standardAddress = addHexPrefix(customAddress).toLowerCase();
 
+    let isMainnetToken = false;
+
+    Object.keys(contractMap).forEach((key) => {
+      if (key.toLowerCase() === customAddress.toLowerCase()) {
+        isMainnetToken = true;
+      }
+    });
```
Comment (danjm) [inline ui/pages/import-token/import-token.component.js:216]: Nit: this code can be simplified with the `.some` method

```suggestion
    const isMainnetToken = Object.keys(contractMap).some((key) => key.toLowerCase() === customAddress.toLowerCase());
```

File: ui/pages/import-token/import-token.component.js [hunk 2]
```diff
@@ -133,6 +135,12 @@ class ImportToken extends Component {
     );
   }
 
+  hasWarning() {
```
Comment (danjm) [inline ui/pages/import-token/import-token.component.js:138]: It looks like this method is not used... can it be deleted?

**Review by danjm — DISMISSED** [review]
This looks like a good solution. I have left a couple of small comments

File: ui/pages/import-token/import-token.component.js (hunk 2 above)
Comment (dragana8) [reply ui/pages/import-token/import-token.component.js]: deleted

File: ui/pages/import-token/import-token.component.js (hunk 1 above)
Comment (dragana8) [reply ui/pages/import-token/import-token.component.js]: updated

**Comment by darkwing** [comment]
The code looks good but I feel like from a design perspective, we should probably do more to inform the user about why this may not work.  Let me see what design thinks.

File: app/_locales/en/messages.json [hunk 3]
```diff
@@ -1576,6 +1576,9 @@
   "mainnet": {
     "message": "Ethereum Mainnet"
   },
+  "mainnetToken": {
+    "message": "This is a mainnet token"
```
Comment (danjm) [inline app/_locales/en/messages.json:1580]: Perhaps to resolve @darkwing's point about design, we could just modify this text a bit. Change it to "This is a mainnet token address. Adding it to the current network is likely to fail."

**Review by danjm — DISMISSED** [review]
Approving, but we could make the one copy change I suggested, and I think that would address @darkwing's design concern as well.

File: app/_locales/en/messages.json (hunk 3 above)
Comment (dragana8) [reply app/_locales/en/messages.json]: Done.

**Review by danjm — DISMISSED** [review]
LGTM

File: app/_locales/en/messages.json [hunk 4]
```diff
@@ -1594,6 +1594,9 @@
   "mainnet": {
     "message": "Ethereum Mainnet"
   },
+  "mainnetToken": {
+    "message": "This is a mainnet token address. We do not expect it to work on the current network..."
```
Comment (adonesky1) [inline app/_locales/en/messages.json:1598]:
```suggestion
    "message": "This address matches a known mainnet token address. Recheck that you are on the right network and the contract address for the token you are trying to add..."
```
cc @rachelcope @danjm I think this copy is more clear. What do you think?

Comment (danjm) [reply app/_locales/en/messages.json:1598]: looks good to me

**Review by adonesky1 — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**